File: problems/1759_count_number_of_homogenous_substrings/Solution.java
class Solution {
    public int countHomogenous(String s) {
        long MOD = 1_000_000_007, result = 0, count = 1;
        for (int i = 1; i < s.length(); i++) {
            if (s.charAt(i) == s.charAt(i - 1)) count++;
            else { result += count * (count + 1) / 2; count = 1; }
        }
        result += count * (count + 1) / 2;
        return (int)(result % MOD);
    }
}

File: problems/1759_count_number_of_homogenous_substrings/solution.py
class Solution:
    def countHomogenous(self, s: str) -> int:
        MOD = 10 ** 9 + 7
        result = 0
        count = 1
        for i in range(1, len(s)):
            if s[i] == s[i - 1]:
                count += 1
            else:
                result += count * (count + 1) // 2
                count = 1
        result += count * (count + 1) // 2
        return result % MOD
